sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultMethodHandler.java

```diff
@@ -96,14 +97,11 @@ private static Method privateLookupIn() {
     }
 
     @Override
-    @SuppressWarnings("PMD.AvoidRethrowingException")
     public Object invoke(Object target, Object[] args, HandleSupplier handle) {
         try {
             return methodHandle.bindTo(target).invokeWithArguments(args);
-        } catch (RuntimeException | Error e) {
-            throw e;
-        } catch (Throwable throwable) {
-            throw new RuntimeException(throwable);
+        } catch (Throwable t) {
+            throw Throwables.uncheck(t);
```

Comment:
Then we'll need to add more catch statements to separate them, which kinda defeats the point of this method... I agree we don't wanna wrap Errors, but catching Error separately so any Throwable left will be an Exception, just adds more boilerplate and a pitfall to remember instead of simplifying things...

I've thought about this before and basically there is no good way to do it, except for the way Unchecked works in jool: instead of transforming exceptions, wrap exception-throwing runnables (and other functionalinterfaces) in a re-usable handler: `Bar bar = Unchecked.function(Foo::toBar).accept(foo).`

Copying Unchecked means stepping into legal argument territory though... apache 2.0 license and whatnot. I personally think a simple Throwable sieve function is so trivial it can't fall under copy protection any more than overriding tostring/hashcode/equals can, but I'm no lawyer...

Reply:
It looks like `jool` is Apache 2 same as `jdbi` -- so what licensing issue are you worried about?

Reply:
There is no license conflict. Jdbi is Apache 2.0 license, and so is Jool, but that's beside the point.

It is generally understood in Java that Throwable has "only" two direct subtypes: Error (for unrecoverable problems like out of memory), and Exception for recoverable problems.

It is possible to extend Throwable directly but this is widely frowned upon.

I know there are a few classes in the wild that extend Throwable instead of Exception, but they are the exception (ha) rather than the rule. This is not an edge case I'm willing to accommodate if we have to ugly up our code to do it.

Side note, this is probably why the old Guava `Throwables.propagate()` method threw the exception for you--to avoid the `RuntimeException | Error` disjunct exception type problem.

Reply:
I don't know what you need to do beside copying the code to be permitted to copy it.

As for subtypes: the method should accept Throwable because some APIs declare it that way, like invocations of methodhandles.